src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -61,7 +61,53 @@ void GcsAutoscalerStateManager::HandleReportAutoscalingState(
 
 void GcsAutoscalerStateManager::GetPendingGangResourceRequests(
     rpc::autoscaler::GetClusterResourceStateReply *reply) {
-  throw std::runtime_error("Unimplemented");
+  // Get the gang resource requests from the placement group load.
+  auto placement_group_load = gcs_resource_manager_.GetPlacementGroupLoad();
+  if (!placement_group_load) {
+    return;
+  }
+
+  // Iterate through each placement group load.
+  for (const auto &pg_data : placement_group_load->placement_group_data()) {
+    auto gang_resource_req = reply->add_pending_gang_resource_requests();
+    // For each placement group, if it's not pending/rescheduling, skip it since.
+    // it's not part of the load.
+    if (!(pg_data.state() == rpc::PlacementGroupTableData::PENDING ||
+          pg_data.state() == rpc::PlacementGroupTableData::RESCHEDULING)) {
+      continue;
+    }
+
+    // Default no anti-affinity for PG.
+    absl::optional<rpc::PlacementConstraint> anti_affinity_constraint = absl::nullopt;
```

Comment:
i think we need also handle strict_pack.

Reply:
think more, we also need to handle partial update  @rkooo567 does it already covered in pg_manager, or we need to handle it here?

Reply:
> i think we need also handle strict_pack.

something like affinity constraint?

Reply:
Hmm actually I don't know how this works for retry case. We may need some experimentation to verify how this works.